server/server_test.go

```diff
@@ -195,4 +217,35 @@ func TestHTTPServer_ReportEndpoint(t *testing.T) {
 	}, testTimeout)
 }
 
+func TestHTTPServer_ClusterReportEndpoint(t *testing.T) {
```

Comment:
Maybe worth renaming TestHTTPServer_SingleRuleEndpoint? Cuz it confused me that it's the same test as `TestHTTPServer_ReportEndpoint`. But it's a nitpick :)